### ISSUE TITLE
 FunctionNaming rule false positives

### ISSUE THREAD

**Issue by jinbinyang** [detekt/detekt#7784]
## Expected Behavior
Code scanning shows no issues.

## Observed Behavior
Property 'style>FunctionOnlyReturningConstant>excludedFunctions' should be a YAML array instead of a comma-separated String.
Property 'style>FunctionOnlyReturningConstant>excludeAnnotatedFunction' is deprecated. Use `ignoreAnnotated` instead.
Property 'style>UseDataClass>excludeAnnotatedClasses' is deprecated. Use `ignoreAnnotated` instead.
Property 'naming>FunctionParameterNaming>ignoreOverridden' is deprecated. This configuration is ignored and will be removed in the future.
Property 'complexity>ComplexMethod' is deprecated. Rule is renamed to `CyclomaticComplexMethod` to distinguish between Cyclomatic Complexity and Cognitive Complexity.
Property 'style>OptionalWhenBraces' is deprecated. Same functionality is implemented in BracesOnWhenStatements.
Property 'naming>ConstructorParameterNaming>ignoreOverridden' is deprecated. This configuration is ignored and will be removed in the future.
Property 'style>LibraryEntitiesShouldNotBePublic' is deprecated. Rule migrated to `libraries` ruleset plugin.
Property 'potential-bugs>RedundantElseInWhen' is deprecated. Rule deprecated as compiler performs this check by default.
Property 'style>ForbiddenPublicDataClass' is deprecated. Rule migrated to `libraries` ruleset plugin.
Property 'style>LibraryCodeMustSpecifyReturnType' is deprecated. Rule migrated to `libraries` ruleset plugin.
Property 'potential-bugs>DuplicateCaseInWhenExpression' is deprecated. Rule deprecated as compiler performs this check by default.
Property 'potential-bugs>MissingWhenCase' is deprecated. Rule deprecated as compiler performs this check by default.
Property 'style>UnnecessaryAbstractClass>excludeAnnotatedClasses' is deprecated. Use `ignoreAnnotated` instead.
Property 'naming>VariableNaming>ignoreOverridden' is deprecated. This configuration is ignored and will be removed in the future.
Property 'style>MandatoryBracesIfStatements' is deprecated. Use `BracesOnIfStatements` with `always` configuration instead.
/data/myTest_code/DetektRuleFunctionNaming.kt:4:1: Unexpected indentation (9) (should be 8) [Indentation]
/data/myTest_code/DetektRuleFunctionNaming.kt:2:5: Function names should match the pattern: [a-z][a-zA-Z0-9]* [FunctionNaming]

formatting - 5min debt
        Indentation - [Unexpected indentation (9) (should be 8)] at /data/myTest_code/DetektRuleFunctionNaming.kt:4:1
naming - 5min debt
        FunctionNaming - [Function names should match the pattern: [a-z][a-zA-Z0-9]*] at /data/myTest_code/DetektRuleFunctionNaming.kt:2:5

## Steps to Reproduce
**Scanned source code：**

> DetektRuleFunctionNaming.kt

```
@Composable
fun fooButton(text: String, onClick: () -> Unit) {
    Button(onClick = onClick) {
        Text(text)
    }
}
```
**Configuration：**
```
  FunctionNaming:           
    active: true           
    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/androidUnitTest/**', '**/androidInstrumentedTest/**', '**/jsTest/**', '**/iosTest/**']
    functionPattern: '[a-z][a-zA-Z0-9]*'
    excludeClassPattern: '$^'
    ignoreAnnotated: ['Composable']
```
**Execute command：**
`java -jar /usr/tool/detekt-cli.jar -p /usr/tool/detekt-formatting.jar -c /usr/config/default-detekt-config-1.23.6.yml -i /data/myTest_code -r xml:/usr/result.xml`

## Context
It seems to indicate that the configuration `ignoreAnnotated: ['Composable']` is ineffective.

## Your Environment

* Version of detekt used:1.23.6
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by BraisGabin** [comment]
I have this rule working with exactly your configuration:

```yaml
  FunctionNaming:
    active: true
    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/androidUnitTest/**', '**/androidInstrumentedTest/**', '**/jsTest/**', '**/iosTest/**']
    functionPattern: '[a-z][a-zA-Z0-9]*'
    excludeClassPattern: '$^'
    ignoreAnnotated:
      - 'Composable'
```

And it works as expected. Are you sure that you are pointing to your configuration file? If you set `active` to `false` the rule is desactived or still raises the same error?

**Comment by jinbinyang** [comment]
I have set the `active `configuration to `false`, and no alerts are generated. When I reconfigure it to `true`, the alert still appears. I have created a new project to store the problematic files. If possible, could you please take a look and see if there is any issue?

link：https://github.com/jinbinyang/detekt_issue_file_FunctionNaming.git

**Comment by jinbinyang** [comment]
I would like to know if there has been any progress on this issue？

**Comment by jinbinyang** [comment]
Is this a bug, or is there something wrong with my configuration?

**Comment by jinbinyang** [comment]
Is there any progress on this issue?

**Comment by cortinico** [comment]
Nope there is not progress. Once there is progress, someone will comment here.

**Comment by jinbinyang** [comment]
I'm still following.

**Comment by atulgpt** [comment]
Not stale

**Comment by atulgpt** [comment]
Hi @jinbinyang in your shared project how to run the detekt task(are you running it via cli?) which is giving you the error?

**Comment by jinbinyang** [comment]
> Hi [@jinbinyang](https://github.com/jinbinyang) in your shared project how to run the detekt task(are you running it via cli?) which is giving you the error?您好，在您的共享项目中，如何运行导致错误的 detekt 任务（您是通过命令行运行的吗？）？

As described, the command I executed is:
java -jar /usr/tool/detekt-cli.jar -p /usr/tool/detekt-formatting.jar -c /usr/config/default-detekt-config-1.23.6.yml -i /data/myTest_code -r xml:/usr/result.xml
